CryptoTechReminderSystem/Gateway/HarvestGateway.cs

```diff
@@ -104,16 +117,30 @@ private JObject RetrieveFromEndPoint(string address)
 
         private JObject RetrieveWithPagination(string address)
         {
-            var apiResponse = RetrieveFromEndPoint($"{address}&page=1");
+            var apiResponse = RetrieveFromEndPoint(AppendURLWithPageNumber(address, 1));
+            
+            if (apiResponse["total_pages"] == null) {
+                Console.WriteLine(apiResponse);
+            }
+
             var totalPages = (int) apiResponse["total_pages"];
 
-            for (var page = 2; page <= totalPages ; page++)
+            for (var pageNumber = 2; pageNumber <= totalPages; pageNumber++)
             {
-                apiResponse.Merge(RetrieveFromEndPoint($"{address}&page={page}"));
+                apiResponse.Merge(RetrieveFromEndPoint(AppendURLWithPageNumber(address, pageNumber)));
             }
 
             return apiResponse;
         }
+
+        private string AppendURLWithPageNumber(string address, int pageNumber)
```

Comment:
I really like this method extraction here, nice one!

Reply:
Ah this was on @C-gyorfi 's recommendation 🤩